server/pike.go

```diff
@@ -18,10 +18,37 @@ func (c *Configuration) pikeNewContact(rc *gin.Context) {
 		return
 	}
 
-	if err = c.pikeActions.AddContact(rc.Request.Context(), receiverPaymail, &requesterContact); err != nil {
+	if err = c.pikeContactActions.AddContact(rc.Request.Context(), receiverPaymail, &requesterContact); err != nil {
 		ErrorResponse(rc, ErrorAddContactRequest, err.Error(), http.StatusExpectationFailed)
 		return
 	}
 
 	rc.Status(http.StatusCreated)
 }
+
+func (c *Configuration) pikeGetPaymentDestinations(rc *gin.Context) {
+	var paymentDestinationRequest paymail.PikePaymentOutputsPayload
+	err := json.NewDecoder(rc.Request.Body).Decode(&paymentDestinationRequest)
+	defer func() {
+		_ = rc.Request.Body.Close()
+	}()
+	if err != nil {
+		ErrorResponse(rc, ErrorInvalidParameter, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	alias, domain, md, ok := c.GetPaymailAndCreateMetadata(rc, paymentDestinationRequest.Amount)
+	if !ok {
+		return
```

Comment:
This method `c.GetPaymailAndCreateMetadata` sets the error response in context. It is a bit unintuitive but I left it in this method because if we wanted to configure it when called, we would have to return 3 error parameters which looks like worse solution in my opinion

Reply:
So, maybe just put a comment here (before the return) 🤔 
...Or just leave like it is :) up to you

Reply:
added comments